stdlib/public/Cxx/CxxDictionary.swift

```diff
@@ -38,4 +38,22 @@ extension CxxDictionary {
       return iter.pointee.second
     }
   }
+  
+  func filter(
+    _ isIncluded: (_ key: Key, _ value: Value) throws -> Bool
+) rethrows -> [Key: Value] {
+    var filteredDictionary: [Key: Value] = [:]
+    var iterator = __findUnsafe(Key) 
```

Comment:
Is there's a way to point at the beginning of the dictionary here ?

Reply:
@egorzhdan hey, are you still active on this PR?

Reply:
Sorry for the delay!
C++ dictionaries in Swift will have `__beginUnsafe()` and `__endUnsafe()` methods (mapped from `begin()` and `end()` methods defined on the C++ side), so you can call `__beginUnsafe()` here. You'll need to declare `__beginUnsafe()` as a member of `CxxDictionary` protocol for that to work.

Reply:
no worries. it's done
Please check it out

Reply:
Just built the project using this command 

`utils/build-script --skip-build-benchmarks \
  --skip-ios --skip-watchos --skip-tvos --swift-darwin-supported-archs "$(uname -m)" \
  --sccache --release-debuginfo --swift-disable-dead-stripping`
  
  but I'm just getting errors related to DAK Layouts.

Reply:
I see. I won't be able to help without seeing the errors unfortunately. Other folks who work with `build-script` a lot should know how to fix this. Could you please post the build log you're seeing on https://forums.swift.org/c/development/compiler/9 and ask how to workaround it?